### PR TITLE
Reverse operation order in QSVT compute_matrix, update tests

### PR DESCRIPTION
**Pull request by DSGuala**
**Context:**
Calling `qml.matrix(qml.qsvt(*args))` gave the incorrect matrix.

**Description of the Change:**
Updated `qml.QSVT.compute_matrix` to use the equation operation order when computing the matrix.

**Benefits:**
QSVT now gives the correct matrix

### CODE REVIEW

File: tests/templates/test_subroutines/test_qsvt.py [hunk 1]
```diff
@@ -72,11 +72,11 @@ def test_init_error(self):
                 [qml.RZ(-2 * theta, wires=0) for theta in [1.23, -0.5, 4]],
                 [0],
                 [
-                    qml.RZ(-8, wires=0),
+                    qml.RZ(-2.46, wires=0),
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qsvt.py:75]: Why this test should be changed?

File: pennylane/templates/subroutines/qsvt.py [hunk 2]
```diff
@@ -371,7 +371,7 @@ def compute_matrix(*args, **kwargs):
                     op_list.append(adjoint(UA_copy))
 
             op_list.append(projectors[-1])
-            mat = qml.matrix(qml.prod(*tuple(op_list)))
+            mat = qml.matrix(qml.prod(*tuple(op_list[::-1])))
```
Comment (trbromley) [inline pennylane/templates/subroutines/qsvt.py:374]: Is this the solution to the issue @Jaybsoni was mentioning yesterday?

File: tests/templates/test_subroutines/test_qsvt.py (hunk 1 above)
Comment (DSGuala) [reply tests/templates/test_subroutines/test_qsvt.py:75]: The order of the operations was backwards.
It didn't fail before because the operations commute (?).

File: tests/templates/test_subroutines/test_qsvt.py [hunk 3]
```diff
@@ -309,33 +309,31 @@ class Testqsvt:
...
             (
                 [[0.1, 0.2], [0.3, 0.4]],
                 [0.2, 0.3],
                 [0, 1],
-                [
-                    qml.PCPhase(0.3, dim=2, wires=[0, 1]),
-                    qml.BlockEncode([[0.1, 0.2], [0.3, 0.4]], wires=[0, 1]),
-                    qml.PCPhase(0.2, dim=2, wires=[0, 1]),
-                ],
+                # mathematical order of gates:
```
[the remark this answers is not in the document]
Comment (DSGuala) [reply tests/templates/test_subroutines/test_qsvt.py:318]: That's correct.

[the remark this answers is not in the document]
Comment (DSGuala) [reply tests/templates/test_subroutines/test_qsvt.py:318]: That's correct.

File: tests/templates/test_subroutines/test_qsvt.py [hunk 4]
```diff
@@ -439,30 +433,20 @@ def test_qsvt_tensorflow(self, input_matrix, angles, wires):
...
-        ("A", "phis", "wires"),
-        [
-            (
-                [[0.1, 0.2], [0.3, 0.4]],
-                [0.1, 0.2, 0.3],
-                [0, 1],
-            )
-        ],
-    )
-    def test_qsvt_grad(self, A, phis, wires):
```
[the remark this answers is not in the document]
Comment (DSGuala) [reply tests/templates/test_subroutines/test_qsvt.py:452]: We were using `pytest.mark.parametrize` to run a single test rather than multiple. So I removed it for simplicity.

File: pennylane/templates/subroutines/qsvt.py (hunk 2 above)
Comment (DSGuala) [reply pennylane/templates/subroutines/qsvt.py:374]: :thinking: I'm not sure. We did discuss the solution in this PR and I believe @Jaybsoni and I are on the same page.

File: tests/templates/test_subroutines/test_qsvt.py (hunk 1 above)
Comment (soranjh) [reply tests/templates/test_subroutines/test_qsvt.py:75]: Then the order in the previous tests should be changed as well? e.g., line 62 and lines 65-67.

Comment (DSGuala) [reply tests/templates/test_subroutines/test_qsvt.py:75]: Lines 62 and 65-67 were in the correct order.

The order in `lst_projectors` and `operations` should be the same. These both refer to circuit order.

File: pennylane/templates/subroutines/qsvt.py (hunk 2 above)
Comment (obliviateandsurrender) [reply pennylane/templates/subroutines/qsvt.py:374]: Yes, previously, without the `[::-1]`, the matrix computed in the `qml.prod` was of the opposite order, which was causing the matrix computation mismatch.

**Review by obliviateandsurrender — APPROVED** [review]
Thanks @DSGuala, looks good! It might be a good idea to check the ordering for the unit-test examples manually once more according to the `qml.prod` decomposition just to be sure.

File: pennylane/templates/subroutines/qsvt.py (hunk 2 above)
Comment (Jaybsoni) [reply pennylane/templates/subroutines/qsvt.py:374]: Yes @trbromley, 

There was a typo in the test/example I showed with the order of the operations. This fixes that and ensures everything produces the same results 👍🏼

**Review by soranjh — APPROVED** [review]

File: tests/templates/test_subroutines/test_qsvt.py [hunk 5]
```diff
@@ -439,30 +433,20 @@ def test_qsvt_tensorflow(self, input_matrix, angles, wires):
...
                 phis,
-                wires=wires,
+                wires=[0, 1],
             )
             return qml.expval(qml.PauliZ(wires=0))
 
-        A = np.array([[0.1j, 0.2], [0.3, 0.4]], dtype=complex, requires_grad=True)
-        phis = np.array([0.1, 0.2, 0.3], requires_grad=True)
+        A = np.array([[0.1, 0.2], [0.3, 0.4]], dtype=complex, requires_grad=True)
+        phis = np.array([0.1, 0.2, 0.3], dtype=complex, requires_grad=True)
```
Comment (Jaybsoni) [inline tests/templates/test_subroutines/test_qsvt.py:449]: Why this change?

**Review by Jaybsoni — APPROVED** [review]

File: tests/templates/test_subroutines/test_qsvt.py (hunk 5 above)
Comment (DSGuala) [reply tests/templates/test_subroutines/test_qsvt.py:449]: Just to make sure it handles complex types :+1: 
Let me know your thoughts if it doesn't make sense.
